client/web/src/api.ts

```diff
@@ -0,0 +1,27 @@
+let csrfToken: string
+
+// apiFetch wraps the standard JS fetch function
+// with csrf header handling.
+export function apiFetch(
+  input: RequestInfo | URL,
+  init?: RequestInit | undefined
+): Promise<Response> {
+  return fetch(input, {
+    ...init,
+    headers: withCsrfToken(init?.headers),
+  }).then((r) => {
+    updateCsrfToken(r)
```

Comment:
if we're only updating the csrf token on api responses, then how is the token set on the initial api request?  I think we need the `csrf.Protect` middleware below on more than just the `/api/` handler... that would allow bootstrapping the csrf token for the first api request.  I'll need to look at the gorilla docs a little more closely to confirm that's how we're supposed to be holding it.